aws/data_source_aws_workspaces_image.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsWorkspacesImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsWorkspacesImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"image_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"os": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"required_tenancy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsWorkspacesImageRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).workspacesconn
+
+	imageID := d.Get("image_id").(string)
+	input := &workspaces.DescribeWorkspaceImagesInput{
+		ImageIds: []*string{aws.String(imageID)},
+	}
+
+	resp, err := conn.DescribeWorkspaceImages(input)
+	if err != nil {
+		return fmt.Errorf("Failed describe workspaces images: %s", err)
```

Comment:
We prefer error messages to use the [Go 1.13 error wrapping verb `%w`](https://blog.golang.org/go1.13-errors#TOC_3.3.)